File: cpp/ISP/step1/Penguin.cpp
#include <string>
#include <exception>
#include "Bird.cpp";
using namespace std;

class Penguin : public Bird {
public:
	Penguin(int initial_feather_count) {
		number_of_feathers_ = initial_feather_count;
	}

	void Molt() override {
		number_of_feathers_ -= 1;
	}

	void Fly() override {
		throw exception("Unsupported Operation Exception");
	}

	void swim() {
		current_location_ = "in the water";
	}

private:
	string current_location_;
	int number_of_feathers_;
};

File: python/ISP/step1/penguin.py
from bird import Bird

class Penguin(Bird):
    def __init__(self, initial_feather_count) -> None:
        super().__init__()
        self.__current_location = ""
        self.__number_of_feathers = initial_feather_count

    def fly(self):
        raise Exception('Unsupported Operation Exception')

    def molt(self):
        self.__number_of_feathers -= 1

    def swim(self):
        self.__current_location = "in the water"
